packages/hatchify-core/src/dataTypes/datetime/getFinalize.ts

```diff
@@ -34,13 +34,85 @@ export function getFinalize(
+    //         : throw "'4 $core' is not a valid date";
+    setClientPropertyValue: (userValue: UserValue): Date | null => {
+      if (typeof userValue === "string") {
+        if (!isISO8601DateString(userValue)) {
+          throw new HatchifyCoerceError("as an ISO 8601 date string")
+        }
+
+        return coerce(new Date(userValue), control)
+      }
+
+      return coerce(userValue, control)
+    },
+
+    // Passed  - a valid "client" version of the data type
+    // Returns - Something that can be serialized to JSON
+    // Example : new Date() => '2023-07-17T01:45:28.778Z'
+    serializeClientPropertyValue: (value: Date | null): string | null => {
+      const coerced = coerce(value, control)
+
+      return coerced === null ? null : coerced.toISOString()
+    },
+
+    // Passed  - Any crazy value the client might use to filter
+    // Returns - A value react-rest can understand / use
+    // Throws  - If the data is bad ❓
+    // Example : ?filter[age]=xyz ... xyz => throw "xyz is not a number";
+    setClientQueryFilterValue: (userValue: UserValue): Date | null => {
+      if (typeof userValue === "string") {
+        if (!isISO8601DateString(userValue)) {
+          throw new HatchifyCoerceError("as an ISO 8601 date string")
+        }
+
+        return coerce(new Date(userValue), control)
+      }
+
+      return coerce(userValue, control)
+    },
+
+    // Passed  - Any value someone might try to query by
+    // Returns - A STRING value that can be sent as part of the query string.
+    // Example : true => "true";
+    serializeClientQueryFilterValue: (value: Date | null): string | null => {
+      const coerced = coerce(value, control)
+
+      return coerced === null ? null : coerced.toISOString()
+    },
+
+    // Passed  - Any crazy STRING value the client might send as GET
+    // Returns - A type the client can use
+    // Throws  - If the data is bad ❓. Also, what should the behavior be?
+    // Example : '2023-07-17T01:45:28.778Z' => new Date('2023-07-17T01:45:28.778Z')
+    setClientPropertyValueFromResponse: (
+      jsonValue: ValueInRequest,
+    ): Date | null => {
+      if (typeof jsonValue === "string") {
+        if (!isISO8601DateString(jsonValue)) {
+          throw new HatchifyCoerceError("as an ISO 8601 date string")
+        }
+
+        return coerce(new Date(jsonValue), control)
+      }
+
+      return coerce(jsonValue, control)
+    },
 
     // Passed  - Any crazy value the client might send as a POST or PATCH
     // Returns - A type the ORM can use
     // Throws  - If the data is bad ❓
     // Example : '2023-07-17T01:45:28.778Z' => new Date('2023-07-17T01:45:28.778Z')
     //         : throw "'4 $core' is not a valid date";
+    // todo: Roye is this incorrect? `ValueInRequest` can be `Date` object,
+    // this won't come in via json. this function fails on string, should
+    // it check if it's a valid iso8601 string first and then coerce?
```

Comment:
@roy-coder thoughts on this comment? `setORMPropertyValue` should be taking in value from api request, therefore it's expected for it to be a string (not date object). this fn should behave similarly to what I have on line 987 in this file in that case?

Reply:
in tests you're using it as a `Date` - but in the real world use case this function will be taking in client value via HTTP request and won't be a `Date` but an (iso8601) `string` no?